### PR TITLE
fixed: target not in cache to stop subscription

### PR DESCRIPTION
**Pull request by fahadnaeemkhan**
see https://github.com/openconfig/gnmic/issues/286 for details

### CODE REVIEW

File: pkg/cache/oc_cache.go [hunk 1]
```diff
@@ -184,8 +184,9 @@ func (gc *gnmiCache) subscribe(ctx context.Context, ro *ReadOpts, ch chan *Notif
 }
 
 func (gc *gnmiCache) handleSingleQuery(ctx context.Context, ro *ReadOpts, ch chan *Notification) {
+	target := ro.Target
```
Comment (karimra) [inline pkg/cache/oc_cache.go:187]: creating a new variable here is not necessary

File: pkg/cache/oc_cache.go [hunk 2]
```diff
@@ -296,6 +303,7 @@ func (gc *gnmiCache) handleSampledQuery(ctx context.Context, ro *ReadOpts, ch ch
 }
 
 func (gc *gnmiCache) handleOnChangeQuery(ctx context.Context, ro *ReadOpts, ch chan *Notification) {
+	target := ro.Target
```
Comment (karimra) [inline pkg/cache/oc_cache.go:306]: creating a new variable here is not necessary

**Review by karimra — CHANGES_REQUESTED** [review]

**Comment by karimra** [comment]
Ended up making the changes myself.
Thanks for catching this and for the PR!

**Review by karimra — APPROVED** [review]

**Comment by fahadnaeemkhan** [comment]
> Ended up making the changes myself. Thanks for catching this and for the PR!

Thanks for merging it !!!

**Pull request merged**